src/main/ruby/truffleruby/core/truffle/polyglot.rb

```diff
@@ -312,6 +312,14 @@ def backtrace_locations
     def backtrace
       backtrace_locations&.map(&:to_s)
     end
+
+    def marshal_dump
+      raise TypeError, "TruffleRuby foreign exceptions cannot be dumped: #{message}"
```

Comment:
```suggestion
      raise TypeError, "Foreign exception cannot be dumped: #{inspect}"
```
That should give more information.